### ISSUE TITLE
app_name field isn't being set

### ISSUE THREAD

**Issue by wa1ker38552** [kivy/plyer#817]
I'm trying to set the app name of the notification but it still shows up as 'Python'

```py
notification.notify(
        title="New Update Available",
        app_name='MyAsus',
        message=get_response(q),
        app_icon="asus.ico", 
        timeout=5
 )
 ```
![image](https://github.com/user-attachments/assets/ab9a08b1-92fa-4f55-a0b6-08a3c2a6d259)

**Comment by motcs** [comment]
Yes, the app_name setting did not function as expected, and the name of the application was still used after packaging

**Comment by motcs** [comment]
The intended effect should be the same as the alarm prompt, but the result is not like this.
![Uploading 微信截图_20240924143807.png…]()

**Comment by motcs** [comment]
![微信截图_20240924144023](https://github.com/user-attachments/assets/500d8fc9-2ad6-49ae-988c-86c3dea1655c)

**Comment by ihmily** [comment]
### ❓ Why doesn't `app_name` show up in the Windows notification?

**Windows does not use any field in the `NOTIFYICONDATA` structure to display the app name in the top-left corner of the notification.** Instead, it uses metadata from the current running `.exe` file, like the `ProductName` in its version resource.

------

### 🧠 Detailed Explanation

In Windows taskbar balloon notifications, the **application name shown in the top-left corner** (e.g., "Python"):

- Is **not controlled by `NOTIFYICONDATA` fields** like `szTip`, `szInfoTitle`, or `app_name`.
- Is pulled from the **metadata of the running executable file**:
  - Specifically from the `.exe` file’s **version resource**, such as `ProductName`.
  - Or sometimes from the executable/module name (e.g., `python.exe`).

Even if you pass a custom `app_name` into `NOTIFYICONDATA`, **Windows ignores it for the purpose of displaying the app name in the notification**.

------

### ✅ Is there a way to customize the application name in the notification?

#### Yes — but only by creating your own `.exe` file.

To control the name shown in the top-left of a Windows notification balloon:

1. Package your Python script as a `.exe` using `pyinstaller` or similar.
2. Add version resource metadata to the `.exe`:
   - Include fields like `ProductName` and `FileDescription`.
3. Windows will then display your custom `ProductName` as the application name in the notification.

📌 Example `version.txt` for use with `pyinstaller`:

```ini
# UTF-8
VSVersionInfo(
  ffi=FixedFileInfo(
    filevers=(1, 0, 0, 0),
    prodvers=(1, 0, 0, 0),
    mask=0x3f,
    flags=0x0,
    OS=0x40004,
    fileType=0x1,
    subtype=0x0,
    date=(0, 0)
  ),
  kids=[
    StringFileInfo([
      StringTable(
        '040904b0',
        [
          StringStruct('CompanyName', 'Your Company'),
          StringStruct('FileDescription', 'Your App Description'),
          StringStruct('ProductName', 'My Application Name'),
        ]
      )
    ]),
    VarFileInfo([VarStruct('Translation', [1033, 1200])])
  ]
)
```

Then build with:

```bash
pyinstaller your_script.py --version-file version.txt
```

------

If you want to truly change the displayed app name, **you need to compile your script into an `.exe` with custom version information**.

For example:

```
from plyer import notification

notification.notify(
    title="Reminder",
    app_icon='python.ico',
    message="This is a notification from Python!",
    app_name="My Application Name"  # Set the application name
)

```

When running the source code directly in the PyCharm development environment, the notification appears as shown below:

<img width="454" height="249" alt="Image" src="https://github.com/user-attachments/assets/cff6a5cc-dc25-4101-ae59-8d6be77d1df6" />

As you can see, the application name in the notification is still displayed as “python”, even though we set a custom app_name `My Application Name` in the code — it did not take effect.

However, when using the same code and packaging it into an .exe file using PyInstaller, the result is as follows:

<img width="454" height="270" alt="Image" src="https://github.com/user-attachments/assets/3118c1ad-2c03-47c0-a815-35ca4e7e4747" />

Now, the notification correctly shows the custom application name, as defined in the .exe file's metadata (e.g., ProductName in the version resource). **This confirms that Windows uses the metadata from the executable itself, rather than the app_name parameter in the code.**
